messaging/src/main/java/org/axonframework/configuration/Components.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.configuration;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.configuration.Component.Identifier;
+
+/**
+ * Wrapper around a {@link Map} of {@link Component Components} stored per {@link Component.Identifier}.
+ * <p>
+ * Provides a cleaner interface to the {@link NewConfigurer} and {@link NewConfiguration} when interacting with the
+ * configured {@code Components}.
+ *
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+public class Components {
+
+    private final Map<Identifier<?>, Component<?>> components = new ConcurrentHashMap<>();
```

Comment:
Do we need `ConcurrentHashMap`? I suppose we don't as the whole NewConfigurer is not thread-safe/not expected to be used concurrently.

Reply:
Thanks!
 I understand the assumption, I think it may be possible, we will see in the future. 
 I think we don't need to discuss it further.

Reply:
Alright, thanks for the conversation too! Never hurts to think about things like this.